### PR TITLE
feat: different star patterns algorithm added

### PR DESCRIPTION
**Pull request by iamfazal**
# Welcome to JavaScript community

[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new JavaScript files are placed inside an existing directory.
* [x] All filenames should use the UpperCamelCase (PascalCase) style.  There should be no spaces in filenames.
     **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains #743 .

### CODE REVIEW

**Comment by iamfazal** [comment]
Hey @raklaptudirm. when you are planning to merge this PR?

**Comment by raklaptudirm** [comment]
@cclauss Please review this pr.

**Comment by iamfazal** [comment]
@raklaptudirm @cclauss guys how much time still do you need to review this single PR?

**Comment by raklaptudirm** [comment]
@cclauss

File: patterns/star-pattern/diamond.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    // printing spaces
+    for (let inner = 1; inner <=outer; inner++)
+        diamond += " ";
+    // printing star
+    for (let inner = 0; inner < 2 * (row - outer) - 1; inner++)
+        diamond += "*";
+
+    diamond += "\n";
+}
+console.log(diamond);
```
Comment (cclauss) [inline patterns/star-pattern/diamond.js:29]: ⛔ means that GitHub expects that each file ends with one and only on carriage return (\n).

**Comment by cclauss** [comment]
I do not do JavaScript so please do not ask me to review these PRs.

**Comment by raklaptudirm** [comment]
Star patterns were not classified as valid educational algorithms. Closed.

**Pull request closed without merging**